src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -718,15 +719,9 @@ REDHAWK_PALEXPORT bool REDHAWK_PALAPI PalStartBackgroundWork(_In_ BackgroundCall
 
 REDHAWK_PALIMPORT bool REDHAWK_PALAPI PalSetCurrentThreadName(const char* name)
 {
-    const int MAX_THREAD_NAME_SIZE = 15;
-    char name_copy[MAX_THREAD_NAME_SIZE + 1];
-    strncpy(name_copy, name, MAX_THREAD_NAME_SIZE);
-    name_copy[MAX_THREAD_NAME_SIZE] = '\0';
-#ifdef __APPLE__
-    pthread_setname_np(name_copy);
-#else
-    pthread_setname_np(pthread_self(), name_copy);
-#endif //__APPLE__
+    int setNameResult = minipal_set_thread_name(pthread_self(), name);
+    (void)setNameResult; // used
+    assert(setNameResult == 0);
```

Comment:
```suggestion
    // Ignore requests to set the main thread name because
    // it causes the value returned by Process.ProcessName to change.
    if ((pid_t)PalGetCurrentOSThreadId() != getpid())
    {
        int setNameResult = minipal_set_thread_name(pthread_self(), name);
        (void)setNameResult; // used
        assert(setNameResult == 0);
    }
```